src/views/AddTimer.tsx

```diff
@@ -0,0 +1,183 @@
+            setRounds(8);
+        } else {
+            setRounds(1);
+        }
+    };
+
+    const renderTimeInput = (label: string, minutes: number, seconds: number, setMinutes: (value: number) => void, setSeconds: (value: number) => void, inputPrefix: string) => (
+        <div className="form-group">
+            <label htmlFor={`${label.toLowerCase().replace(' ', '-')}-minutes`}>{label}</label>
+            <div className="time-input-pair">
+                <div className="time-input">
+                    <label htmlFor={`${inputPrefix}-minutes`}>Minutes:</label>
+                    <input
+                        id={`${inputPrefix}-minutes`}
+                        type="number"
+                        className="input-inline"
+                        placeholder="0"
+                        min="0"
+                        max="59"
+                        value={minutes}
+                        onChange={e => setMinutes(Math.max(0, Number.parseInt(e.target.value || '0')))}
+                    />
+                </div>
+                <div className="time-input">
+                    <label htmlFor={`${inputPrefix}-seconds`}>Seconds:</label>
+                    <input
+                        id={`${inputPrefix}-seconds`}
+                        type="number"
+                        className="input-inline"
+                        placeholder="30"
+                        min="0"
+                        max="59"
+                        value={seconds}
+                        onChange={e => setSeconds(Math.max(0, Number.parseInt(e.target.value || '0')))}
+                    />
+                </div>
+            </div>
+        </div>
+    );
+
+    return (
+        <div className="timers-container">
+            <div className="add-timer-page">
+                <div className="timer-wrapper">
+                    <h2>Add Timer</h2>
+                    {error && <p className="error-message">{error}</p>}
+                    <form
+                        onSubmit={e => {
+                            e.preventDefault();
+                            handleAddTimer();
+                        }}
+                    >
+                        <div className="form-group">
+                            <label htmlFor="timerType">Timer Type:</label>
+                            <select id="timerType" value={timerType} onChange={handleTimerTypeChange} className="input">
+                                <option value="stopwatch">Stopwatch</option>
+                                <option value="countdown">Countdown</option>
+                                <option value="xy">XY</option>
+                                <option value="tabata">Tabata</option>
+                            </select>
+                        </div>
+
+                        {(timerType === 'stopwatch' || timerType === 'countdown') && renderTimeInput('Duration', minutes, seconds, setMinutes, setSeconds, 'duration')}
```

Comment:
You should create a separate component (that can take `timerType` as one of its props) rather than a render function. This would simplify AddTimer, and push the responsibility of figuring out how to render the input fields to a separate component. It would also help DRY up this code.